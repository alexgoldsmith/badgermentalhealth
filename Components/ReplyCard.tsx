import React, { useEffect, useState } from 'react'
import { StyleSheet, Text, TouchableOpacity, View, Image } from 'react-native'
import getStyles from '../Styling/Styling'
import { Post, Replies } from "../Interfaces/Interfaces";
import updateReply from '../Hooks/updateReply';
import getStorageData from '../Hooks/getStorageData';

const styles = getStyles();

type propType = {
    reply: Replies,
    refresh: Function
}

const ReplyCard: React.FC<propType> = ({reply, refresh}) => {
    const [likes, setLikes] = useState<number>(reply.Likes);
    const [user, setUser] = useState<string>('')
    
    useEffect(() => {
        loadUser()
    }, [])

    const loadUser = async () => {
        const data = await getStorageData('user')
        setUser(data.email)
    }

    const increaseLikes = () => {
        setLikes(likes + 1)
        updateLikes(likes + 1)
    }
    
    const updateLikes = (likes: number) => {
        reply.Likes = likes
        updateReply(reply)
    }
    
    return (
        <View style={[styles.postCard, {justifyContent: 'flex-end', width: 275, marginLeft: 65}]}>
            <Text>{reply.Text}</Text>
            <View style={{display: 'flex', flexDirection: 'row', alignItems: 'center'}}>
                <TouchableOpacity 
                        style={{display: 'flex', flexDirection: 'row', justifyContent: 'center', width: 50}}
                        onPress={increaseLikes}
                >
                    <View style={{marginTop: 0, display: 'flex', alignItems: 'center', flexDirection: 'row'}}>
                        <Image style={styles.like_image} source={require('../assets/like.webp')} /> 
                        <Text style={{fontSize: 20}}> {likes} </Text>
                    </View>
                </TouchableOpacity>
                {(reply.Author === user || reply.Author === 'admin@admin.com')?
                    <TouchableOpacity 
                        style={[styles.loginCard, {backgroundColor: 'red', height: 40, minWidth: '50%', margin: 15, marginLeft: 75}]}
                        onPress={() => {/*deletePost(post._id))*/; refresh() }}
                    >
                        <Text style={{fontSize: 12, color: 'white'}}> Delete This Reply </Text>
                    </TouchableOpacity>
                    : <></>
                }
            </View>
        </View>
    )
}

export default ReplyCard
