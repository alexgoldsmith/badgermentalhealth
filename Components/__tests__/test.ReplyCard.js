// https://jestjs.io/docs/snapshot-testing#snapshot-testing-with-jest
// components/__tests__/Hello.tsx
import React from 'react';
import renderer from 'react-test-renderer';
import ReplyCard from '../ReplyCard';

const mockReply = {
    Text: "Test Reply",
    Author: "Test Reply Author",
    PostId: 8,
    Date: new Date(2021, 23, 7),
    Likes: 3,
}

const mockAuthor = "Test Post Author"

it('renders correctly with defaults', () => {
  const component = renderer
    .create(<ReplyCard reply={mockReply} author={mockAuthor}/>)
    .toJSON();
  expect(component).toMatchSnapshot();
});