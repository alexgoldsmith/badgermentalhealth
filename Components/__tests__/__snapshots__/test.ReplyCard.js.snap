// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "display": "flex",
        "justifyContent": "center",
        "margin": 20,
        "padding": 20,
      },
      Object {
        "alignItems": "flex-start",
        "marginLeft": 0,
        "marginRight": "30%",
        "width": "65%",
      },
    ]
  }
>
  <Text>
    Test Reply
  </Text>
  <Text>
    Author: 
    Test Reply Author
  </Text>
  <Text>
    Likes: 
    3
  </Text>
</View>
`;
